### PR TITLE
mongodb 3.1.0

### PR DESCRIPTION
**Pull request by ehershey**

### CODE REVIEW

**Comment by ehershey** [comment]
I'll look at these errors in audit and install tests.. 

http://bot.brew.sh/job/Homebrew%20Pull%20Requests/23038/version=mountain_lion/testReport/junit/brew-test-bot/mountain_lion/install___devel_mongodb/
Checking for libstdc++ 4.8.2 or better... no
When using libstdc++, MongoDB requires libstdc++ 4.8.2 or newer

http://bot.brew.sh/job/Homebrew%20Pull%20Requests/23038/version=mountain_lion/testReport/junit/brew-test-bot/mountain_lion/audit_mongodb/
 * Stable resource "github.com/mongodb/mongo-tools": Git should specify :revision when a :tag is specified.
 * Devel resource "github.com/mongodb/mongo-tools": Git should specify :revision when a :tag is specified.

**Comment by tdsmith** [comment]
It would be more idiomatic to do something like:

```
stable do
   url "foo"
   sha1 "bar"
   go_resource "github.com/mongodb/mongo-tools" do
      url..., :tag => "v3.0.1", :revision => ...
   end
end

devel do
  url...
  sha1...
  go_resource "github.com/mongodb/mongo-tools" do
    url..., :tag => "v3.1.0", :revision => ...
  end
end
```

where you just define the mongo-tools resource differently on the stable and devel blocks.

**Comment by mikemcquaid** [comment]
> It would be more idiomatic to do something like:

Yup :+1:

**Comment by ehershey** [comment]
I redid the change to what I think should work and now I'm getting this error. Any idea how to fix it?

Error: undefined method `go_resource' for #<SoftwareSpec:0x105c409b8>

File: Library/Formula/mongodb.rb [hunk 1]
```diff
@@ -21,6 +31,17 @@ class Mongodb < Formula
...
   end
 
+  devel do
+    url "https://fastdl.mongodb.org/src/mongodb-src-r3.1.0.tar.gz"
+    sha256 "e6c6f32a37fbf9714fa776a6f79a316bd336f5a7e4c96af9ebef8eaf91758744"
+  end
+
+  go_resource "github.com/mongodb/mongo-tools" do
+    url "https://github.com/mongodb/mongo-tools.git",
+      :tag => tools_tag
```
Comment (tdsmith) [inline Library/Formula/mongodb.rb]: `tools_tag` is undefined

**Comment by tdsmith** [comment]
Hmm, we should consider supporting that syntax, sorry. Opened #38330.

**Comment by tdsmith** [comment]
That's landed so that syntax should work once the `tools_tag` reference is changed. Sorry for the wait.

**Comment by ehershey** [comment]
Awesome, thank you! 3.1.1 is already out so I'm going to do a separate pull request.

**Pull request closed without merging**